pkg/asset/installconfig/aws/metadata.go

```diff
@@ -134,22 +134,19 @@ func (m *Metadata) populateSubnets(ctx context.Context) error {
 	m.privateSubnets = sb.Private
 	m.publicSubnets = sb.Public
 	m.edgeSubnets = sb.Edge
-	return err
+	return nil
 }
 
 // VPC retrieves the VPC ID containing PublicSubnets and PrivateSubnets.
 func (m *Metadata) VPC(ctx context.Context) (string, error) {
-	m.mutex.Lock()
-	defer m.mutex.Unlock()
-
```

Comment:
I don't think we should get rid of this. We need to make sure we have exclusive access to the object before reading any value, otherwise we can read a value while it's being set by another routine and therefore get inconsistent results.